app/src/main/java/agency/five/codebase/android/movieapp/ui/favorites/FavoritesViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package agency.five.codebase.android.movieapp.ui.favorites
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.ui.favorites.mapper.FavoritesMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class FavoritesViewModel(
+    private val movieRepository: MovieRepository,
+    private val favoritesMapper: FavoritesMapper,
+) : ViewModel() {
+
+    private val _favoritesViewState = MutableStateFlow(FavoritesViewState())
+    val favoritesViewState = _favoritesViewState.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            movieRepository.favoriteMovies().collect {
+                _favoritesViewState.value = favoritesMapper.toFavoritesViewState(it)
+            }
+        }
+    }
```

Comment:
You don't need to fix this now, but instead of collecting value and just emiting it to another flow, you can just set `favoritesViewState` to be value from `movieRepository.favoriteMovies()` mapped to `FavoritesViewState`. The same can be applied in `MovieDetailsViewModel` and `HomeViewModel`. 
Just try to do it like that in a project.